classes/local/util/manager.php

```diff
@@ -207,6 +213,100 @@ protected function prepare_files_skeletons() {
         if ($this->should_have('events')) {
             $this->prepare_events();
         }
+
+    }
+
+    /**
+     * Prepares the files for an activity module plugin.
+     */
+    protected function prepare_mod_files() {
+
+        $componentname = $this->recipe['component_name'];
+
+        $stringids = array(
+            $componentname.'name',
+            $componentname.'name_help',
+            $componentname.'settings',
+            $componentname.'fieldset',
+            'missingidandcmid',
+            'modulename',
+            'modulename_help',
+            'modulenameplural',
+            'nonewmodules',
+            'pluginadministration',
+            'view'
+        );
+
+        foreach ($stringids as $stringid) {
+
+            $found = false;
+            if (!empty($this->recipe['strings'])) {
+                foreach ($this->recipe['strings'] as $string) {
+                    if ($string['id'] === $stringid) {
+                        $found = true;
+                        break;
+                    }
+                }
+            }
+
+            if (!$found) {
```

Comment:
Another solution that I used was to add a default value for the string, something like 'String id not set: '.$stringid. To avoid those long error messages when navigating Moodle.

We could also throw an exception, but that seems a bit too excessive.